dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/DelayAppStartupHealthCheckTest.java

```diff
@@ -73,6 +75,69 @@ public void cleanUp() throws Exception {
     }
 
+    @Test
+    public void testReadinessEndpointOnServerStart() throws Exception {
+        server1.setMarkToEndOfLog();
+        server1.stopServer("CWWKE1102W", "CWWKE1105W", "CWMH0052W", "CWMH0053W");
+
+        class StartServerOnThread extends Thread {
+            @Override
+            public void run() {
+                try {
+                    server1.startServer();
+                } catch (Exception e) {
+                    assertTrue("Failure to start server on a seperate thread.", server1.isStarted());
+                }
+            }
+        }
+
+        // Need to ensure the server is not finish starting when readiness endpoint is hit so start the server on a separate thread
+        // Note: this does not guarantee that we hit the endpoint during server startup, but it is highly likely that it will
+        StartServerOnThread startServerThread = new StartServerOnThread();
+        log("testReadinessEndpointOnServerStart", "Starting DelayedHealthCheck server on separate thread.");
+        startServerThread.start();
+
+        // Hit the readiness endpoint as the server is still starting up
+        // We expect a connection refused as the ports are not open until server is fully started
+        try {
+            log("testReadinessEndpointOnServerStart", "Testing the /health/ready endpoint as the server is still starting up.");
+            HttpURLConnection conReady = null;
+            String responseCode = null;
+            boolean connectionExceptionEncountered = false;
+
+            try {
+                conReady = HttpUtils.getHttpConnectionWithAnyResponseCode(server1, READY_ENDPOINT);
+                responseCode = conReady.getURL().toString();
+            } catch (ConnectException e) {
+                log("testReadinessEndpointOnServerStart", e.getMessage()); /// REMOVE THIS AFTER
+                if (e.getMessage().contains("Connection refused")) {
+                    connectionExceptionEncountered = true;
+                }
+            }
+
+            startServerThread.join();
+
+            String failure_message = "The connection was not refused as required, but instead completed with response code: " + responseCode +
+                                     " This is likely due to a rare timing issue where the server starts faster than we can hit the readiness endpoint."
+                                     + "In that case there are no issues with the feature and this failure may be disregarded.";
+            assertTrue(failure_message, conReady == null && connectionExceptionEncountered);
+        } catch (Exception e) {
+            startServerThread.join();
+            fail("Encountered an error while Testing the /health/ready endpoint as the server is still starting up. Error: " + e);
+        }
+
+        // Wait until the application is fully started
+        String line = server1.waitForStringInLogUsingMark("(CWWKZ0001I: Application DelayedHealthCheckApp started)+", 60000);
```

Comment:
we should not be doing any log scraping to determine app readiness -- we should be getting that information from the `/health/ready` endpoint instead. We should poll `/health/ready` repeatedly until we get the HTTP 200, and once we get the HTTP 200 we should immediately try to access an application endpoint and it should be accessible